acceptance/tests/validate_vendored_ruby.rb

```diff
@@ -0,0 +1,54 @@
+require 'puppet/acceptance/temp_file_utils'
+extend Puppet::Acceptance::CommandUtils
+
+test_name 'PA-1319: Validate that the vendored ruby can load gems and is configured correctly' do
+
+  step "'gem list' displays a decent list of gems" do
+    minimum_number_of_gems = 3
+    # \d+\.\d+\.\d+ describes the gem version
+    gem_re = /[\w-]+ \(\d+\.\d+\.\d+(?:, \d+\.\d+\.\d+)*\)/
+
+    agents.each do |agent|
+      gem = gem_command(agent) 
+      listed_gems = on(agent, "#{gem} list").stdout.chomp.scan(gem_re)
+      assert(listed_gems.size >= minimum_number_of_gems, "'gem list' fails to list the available gems")
```

Comment:
I'd find it more helpful here if the error message was a bit more verbose about the context of this test. We're running it to make sure the gem environment is sane, so maybe update the error message to explain that the ruby gem environment doesn't appear to be working, or rename the test step name to explain that the test is intended to validate the ruby gem environment is working. When a test case blows up I find it helpful to get some context as to the purpose behind the test.